apps/frontend/src/components/global/upload_tabs/LoadFileList.vue

```diff
@@ -35,26 +82,100 @@
 <script lang="ts">
 import Vue from 'vue';
 import Component from 'vue-class-component';
+import EditEvaluationModal from '@/components/global/upload_tabs/EditEvaluationModal.vue'
 
+import {SnackbarModule} from '@/store/snackbar';
+import {EvaluationModule} from '@/store/evaluations'
 import {IEvaluation} from '@heimdall/interfaces';
 import {Prop} from 'vue-property-decorator';
-import {Samples} from 'aws-sdk/clients/devicefarm';
+import {Sample, Samples} from 'aws-sdk/clients/devicefarm';
+import { Evaluation } from '../../../types/models';
 
 @Component({
-  components: {}
+  components: {
+    EditEvaluationModal
+  }
 })
 export default class LoadFileList extends Vue {
   @Prop({required: true}) readonly headers!: Object[];
   @Prop({type: Boolean, default: false}) readonly loading!: boolean;
   @Prop({type: String, default: 'id'}) readonly fileKey!: string;
+  @Prop({type: String, default: 'filename'}) readonly sortBy!: string;
   @Prop({required: true}) readonly files!: IEvaluation[] | Samples[];
 
   selectedFiles: IEvaluation[] | Samples[] = [];
+  activeIndex: number = -1;
+  activeItem: Evaluation | Sample = {};
+
+  editDialog: boolean = false;
+  deleteDialog: boolean = false;
+  search: string = '';
 
   load_results(evaluations: IEvaluation[]) {
     this.selectedFiles = [];
     this.$emit('load-results', evaluations);
   }
+
+  closeEditModal() {
+    this.editDialog = false;
+  }
+
+  updateEvaluations() {
+    this.$emit('updateEvaluations')
+  }
+
+  editItem (item: IEvaluation) {
+    EvaluationModule.setActiveEvaluation(item);
+    this.editDialog = true;
+  }
+
+  deleteItem(item: any) {
+    EvaluationModule.setActiveEvaluation(item);
+    this.deleteDialog = true;
+  }
```

Comment:
It makes sense to me to store all of the Evaluations in the Evaluation Vuex store. It seems a little strange to save the active one there though. It makes a bit more sense to me for this to be stored in the LoadFileList component itself and then pass the active file to the EditEvaluationModal just like `this.editDialog` and `this.deleteDialog`.

Reply:
While possible, I don't think that this would work too well. It would mean directly mutating what's passed down to EditEvaluationModal (and the way around that would be using emitters, and then handling all edits in LoadFileList)